test/broadcast.jl

```diff
@@ -0,0 +1,157 @@
+# issue 191
+@testset "Broadcast sizes" begin
+    @test broadcast_sizes(1, 1, 1) === (Size(), Size(), Size())
+    for t in (SVector{2}, MVector{2}, SMatrix{2, 2}, MMatrix{2, 2})
+        @test broadcast_sizes(ones(t), ones(t), ones(t)) === (Size(t), Size(t), Size(t))
+        @test broadcast_sizes(ones(t), 1, ones(t)) === (Size(t), Size(), Size(t))
+        @test broadcast_sizes(1, ones(t), ones(t)) === (Size(), Size(t), Size(t))
+        @test broadcast_sizes(ones(t), ones(t), 1) === (Size(t), Size(t), Size())
+        @test broadcast_sizes(1, ones(t), 1) === (Size(), Size(t), Size())
+        @test broadcast_sizes(ones(t), 1, 1) === (Size(t), Size(), Size())
+        @test broadcast_sizes(1, 1, ones(t)) === (Size(), Size(), Size(t))
+    end
+    @test broadcast((a,b,c)->0, SVector(1,1), 0, 0) == SVector(0, 0)
+end
+
+@testset "Broadcast" begin
+    @testset "AbstractArray-of-StaticArray with scalar math" begin
+        v = SVector{2,Float64}[SVector{2,Float64}(1,1)]
+        @test @inferred(v .* 1.0)::typeof(v) == v
+        @test @inferred(1 .- v)::typeof(v) == v .- v
+        v2 = SVector{2,Int}[SVector{2,Int}(1,1)]
+        @test @inferred(v2 .* 1.0)::typeof(v) == v
+    end
+
+    @testset "2x2 StaticMatrix with StaticVector" begin
+        m = @SMatrix [1 2; 3 4]
+        v = SVector(1, 4)
+        @test @inferred(broadcast(+, m, v)) === @SMatrix [2 3; 7 8]
+        @test @inferred(m .+ v) === @SMatrix [2 3; 7 8]
+        @test @inferred(v .+ m) === @SMatrix [2 3; 7 8]
+        @test @inferred(m .* v) === @SMatrix [1 2; 12 16]
+        @test @inferred(v .* m) === @SMatrix [1 2; 12 16]
+        @test @inferred(m ./ v) === @SMatrix [1 2; 3/4 1]
+        @test @inferred(v ./ m) === @SMatrix [1 1/2; 4/3 1]
+        @test @inferred(m .- v) === @SMatrix [0 1; -1 0]
+        @test @inferred(v .- m) === @SMatrix [0 -1; 1 0]
+        @test @inferred(m .^ v) === @SMatrix [1 2; 81 256]
+        @test @inferred(v .^ m) === @SMatrix [1 1; 64 256]
+    end
+
+    @testset "2x2 StaticMatrix with 1x2 StaticMatrix" begin
+        m1 = @SMatrix [1 2; 3 4]
+        m2 = @SMatrix [1 4]
+        # @test @inferred(broadcast(+, m1, m2)) === @SMatrix [2 6; 4 8] #197
```

Comment:
All the places where tests are currently commented out (because they're broken) you can use `@test_broken` instead.  This runs the test and verifies that it is, indeed, broken - so we won't forget to add them back in later when these issues are fixed.